docs/generator.py

```diff
@@ -257,6 +284,152 @@ def _generate_from_template(self, dest_dir, filename):
+            self._tags_being_described.append(display_name)
+
+        self._generate_resources_config_file(base_dest_dir)
+
+    def _generate_operation_docs(self, operations, dest_dir):
+        template = self._jinja_env.get_template(self.OPERATION_TEMPLATE)
+
+        for op_name, op_spec in operations.items():
+            data_params = self._get_data_params(op_name, op_spec)
+            op_content = template.render(
+                name=op_name,
+                description=op_spec.get(OperationField.DESCRIPTION),
+                method=op_spec.get(OperationField.METHOD),
+                url=op_spec.get(OperationField.URL),
+                path_params=op_spec.get(OperationField.PARAMETERS, {}).get(OperationParams.PATH, {}),
+                query_params=op_spec.get(OperationField.PARAMETERS, {}).get(OperationParams.QUERY, {}),
+                data_params=data_params,
+                curl_sample=swagger_ui_curlify.curlify(
+                    op_spec, data_params, self._api_spec[SpecProp.MODELS], BASE_HEADERS),
+                bravado_sample=swagger_ui_bravado.generate_sample(
+                    op_name, op_spec, data_params, self._api_spec[SpecProp.MODELS]
+                ),
+                **self._template_ctx
+            )
+
+            self._write_generated_file(dest_dir, op_name + self.MD_SUFFIX, op_content)
+
+    def _generate_overview_docs(self, tag_name, operations, dest_dir):
+        template = self._jinja_env.get_template(self.OVERVIEW_TEMPLATE)
+        # model_api_spec = self._api_spec[SpecProp.MODELS].get(model_name, {})
+        content = template.render(
+            # name=self._get_display_model_name(tag_name),
+            operations=operations.keys(),
+            # description=model_api_spec.get(PropName.DESCRIPTION, {}),
+            # properties=model_api_spec.get(PropName.PROPERTIES, {}),
+            **self._template_ctx
+        )
+        self._write_generated_file(dest_dir, 'overview.md', content)
+
+    def _generate_config_json(self, operations, dest_dir):
+        template = self._jinja_env.get_template(self.CONFIG_TEMPLATE)
+        content = template.render(operations=operations.keys(), **self._template_ctx)
+        self._write_generated_file(dest_dir, 'config.json', content)
+
+    def _generate_resources_config_file(self, base_dest_dir):
+        config_file_name = os.path.join(base_dest_dir, "config.json")
+        self._tags_being_described.sort()
+        items = [{
+            "title": "Model Index",
+            "content": "../models/index.md"
+        }]
+
+        if self._errors_codes_file:
+            items.append({
+                "title": "Error Codes",
+                "content": "../" + self._errors_codes_file
+            })
+
+        items += [
+            {
+                "title": split_operation_names(tag_name),
+                "type": "config",
+                "content": "{}/config.json".format(tag_name)
+            }
```

Comment:
I would rather create a Jinja template for `config.json` file and delegate this logic to it.